lib/created_message_queue_handler.ts

```diff
@@ -492,6 +507,17 @@ export async function index(
               newMessageWithContent.id
             }`
           );
+
+          appInsightsClient.trackEvent({
+            measurements: {
+              elapsed: Date.now() - newMessageWithContent.createdAt.getTime()
```

Comment:
this measurement is not really useful, `createdAt` gets set by `cosmos`, `Date.now()` is this system clock, the two times may not even be correlated, esp. in very short time frames (e.g. `createdAt` may be "after" `Date.now()`) - when measuring performance you need to always sample the clock from the same source

Reply:
this is not the value of the '_ts' parameter of cosmosdb, but a custom value set = Date.now() when creating a new message so it's fine to compare that with Date.now()

Reply:
yes, but same reasoning applies, since that value is saved in the db and read from the clock of a (possibly) different host - what I mean is that: you can do this, but keep in mind that any misalignment in the clock of the different nodes will be reflected in the difference of the timestamps - since we don't know how good is the clock sync of these hosts, we have no idea of the error we have in these measurements

Reply:
yes I agree, but are there any alternatives ? aside, this 'elapsed' value is usually (if not, always) expressed in terms of seconds (> 4sec. from load tests)

Reply:
one alternative could be to let AI set the timestamp (e.g. when it received the event)

what do we need this `elapsed` for in the context of our analysis?

Reply:
that's the time that passes between creating a message (in the web controller) and sending a notification (imho a pretty useful indicator).

what do you mean with "let AI set the timestamp" ?

Reply:
I mean, but I don't know if it's possible, di have AI set the timestamp of events with its own local clock (assuming that AI's clock will be consistent across events).

Reply:
my point about `elapsed` is that, assuming everything works as expected, `elapsed` will be < a few hundreds msecs - given that the smaller unit of measure is seconds, it looks it's pretty useless for our analysis

Reply:
1. AI always set the timestamp for each event, but you cannot "extract" it at runtime and you cant have a graph with a diff in the UI (as metrics are always aggregated scalar values and cannot be dynamic). maybe you can query it someway in the analytics query UI (but not in the metric explorer), otherwise you have to check it manually.

2. the smaller unit of measure is milliseconds (Date.now / getTime), this is what you get:

![image](https://user-images.githubusercontent.com/148224/40613150-a0f5ef78-627d-11e8-9587-ffe15e6d639a.png)

which means that the avergae time to send a notification (every channel) is 14.09 seconds from the time the message is created calling the API (this measure will vary in production)